ext/standard/tests/strings/sprintf_rope_optimization_003.phpt

```diff
@@ -0,0 +1,134 @@
+--TEST--
+Test sprintf() function : Rope Optimization for '%d'.
+--FILE--
+<?php
+function func($num) {
+	return $num + 1;
+}
+function sideeffect() {
+	echo "Called!\n";
+	return "foo";
+}
+class Foo {
+	public function __construct() {
+		echo "Called\n";
+	}
+}
+
+$a = 42;
+$b = -1337;
+$c = 3.14;
```

Comment:
Memo to myself, it might make sense to change the sprintf family of functions to [emit the deprecation about implicit truncation for floats](https://wiki.php.net/rfc/implicit-float-int-deprecate).